front/lib/api/vaults.ts

```diff
@@ -0,0 +1,117 @@
+import { ConnectorsAPI, CoreAPI, Ok } from "@dust-tt/types";
+
+import config from "@app/lib/api/config";
+import type { DataSourceResource } from "@app/lib/resources/data_source_resource";
+import logger from "@app/logger/logger";
+
+export const getManagedDataSourceContent = async (
+  connectorId: string,
+  permission: "read" | "write" | "read_write" | "none",
+  parentIds: string[] | null,
+  viewType: "tables" | "documents"
+) => {
+  const connectorsAPI = new ConnectorsAPI(
+    config.getConnectorsAPIConfig(),
+    logger
+  );
+
+  const contentNodes = [];
+  if (!parentIds) {
+    contentNodes.push(
+      await connectorsAPI.getConnectorPermissions({
+        connectorId,
+        filterPermission: permission,
+        viewType,
+      })
+    );
+  } else {
+    for (const parentId of parentIds) {
+      contentNodes.push(
+        await connectorsAPI.getConnectorPermissions({
+          connectorId,
+          parentId,
+          filterPermission: permission,
+          viewType,
+        })
+      );
+    }
+  }
+  const err = contentNodes.find((r) => r.isErr());
+  if (err?.isErr()) {
+    return err;
+  }
+
+  const results = contentNodes.flatMap((r) =>
+    r.isOk()
+      ? r.value.resources.map((r) => ({
+          internalId: r.internalId,
+          parentInternalId: r.parentInternalId,
+          type: r.type,
+          title: r.title,
+          expandable: r.expandable,
+          preventSelection: r.preventSelection,
+          dustDocumentId: r.dustDocumentId,
+          lastUpdatedAt: r.lastUpdatedAt,
+        }))
+      : []
+  );
+
+  return new Ok(results);
+};
+
+export const getUnmanagedDataSourceContent = async (
+  dataSource: DataSourceResource,
+  parentIds: string[] | null,
+  viewType: "tables" | "documents",
+  limit: number,
+  offset: number
```

Comment:
i'm not sure to understand this ?

Reply:
To avoid messing up `limit` and `offset` at callsites, we usually go with an object instead so it requires the full param name. This is a pattern we use for natives types (like string and number).

So instead of `myFunction(limit: number, offset: number)` we will do something like `myFunction({limit, offset}: {limit: number, offset: number}`.